### PR TITLE
Adding run-remotely.sh script 

### PR DESCRIPTION
**Pull request by alexanderConstantinescu**
This PR introduces a `run-remotely.sh` script which allows us to run the CNO against a remotely running cluster. It's based off of `run-locally.sh` and takes only a kubeconfig argument. It strips off some of the logic surrounding manifests, install-config, etc that the `run-locally.sh` script has. 

I am not familiar with the reasons behind having the `PLUGIN_IMAGE` variable. But I kept it, in case someone was using it to modify the `env.sh` using that mechanism.

### CODE REVIEW

**Comment by danwinship** [comment]
> This PR introduces a `run-remotely.sh` script which allows us to run the CNO against a remotely running cluster.

That's what `run-locally.sh` is for... (The CNO binary is local, but the cluster is remote.) Rather than having a separate script, it would make more sense to just modify run-locally's "attach" mode to be able to work with a kubeconfig directly rather than only a cluster dir.

**Comment by dcbw** [comment]
FWIW, PLUGIN_IMAGE is for if you have a custom ovnkube or SDN docker image that you want to run in the cluster to test changes. It's less useful now that we have infrastructure for running CI with ovnkube (which we didn't have at the time). But perhaps still useful.

**Comment by alexanderConstantinescu** [comment]
/test e2e-gcp

flake, flake, flake....man, what is it snowing?!

File: hack/run-locally.sh [hunk 1]
```diff
@@ -13,13 +13,21 @@ if [[ -n "${HACK_MINIMIZE:-}" ]]; then
...
+    wait_for_cluster
+    stop_deployed_operator
+    extract_environment_from_running_cluster
+}
+
 # Install our overrides so the cluster doesn't run the network operator.
 function override_install_manifests() {
     # Patch the CVO to not create the network operator
-    kubectl --kubeconfig=hack/null-kubeconfig patch --type=json --local=true -f="${CLUSTER_DIR}/manifests/cvo-overrides.yaml" -p "$(cat hack/overrides-patch.yaml)" -o yaml > "${CLUSTER_DIR}/manifests/.cvo-overrides.yaml.new"
+    oc --kubeconfig=hack/null-kubeconfig patch --type=json --local=true -f="${CLUSTER_DIR}/manifests/cvo-overrides.yaml" -p "$(cat hack/overrides-patch.yaml)" -o yaml > "${CLUSTER_DIR}/manifests/.cvo-overrides.yaml.new"
```
Comment (danwinship) [inline hack/run-locally.sh]: Maybe split this stuff out into a separate commit

File: hack/run-locally.sh [hunk 2]
```diff
@@ -158,15 +164,20 @@ function create_cluster() {
...
 cluster-network-operator will be stopped, plugin image overrides applied, and
 a local cluster-network-operator started.
 
 The following options are accepted for both new and existing clusters:
 
- -c DIR           the cluster installation temporary directory (can also be given via CLUSTER_DIR); required
+ -c DIR           the cluster installation temporary directory (can also be given via CLUSTER_DIR); 
+ -k KUBECONFIG    the kubeconfig pointing to the existing cluster (can also be given via KUBECONFIG); 
+
+-c or -k are required
```
Comment (danwinship) [inline hack/run-locally.sh]: "Either -c or -k is required"
or really
"Either -c or -k (or their corresponding environment variable) is required"

File: hack/run-locally.sh [hunk 3]
```diff
@@ -190,13 +202,17 @@ CLUSTER_DIR="${CLUSTER_DIR:-}"
...
+while getopts "c:f:i:m:k:n:w" opt; do
     case $opt in
-        c) CLUSTER_DIR="${OPTARG}";;
+        c) CLUSTER_DIR="${OPTARG}"
+	   mkdir -p "${CLUSTER_DIR}" >& /dev/null
+	   ;;
         f) INSTALL_CONFIG="${OPTARG}";;
         i) INSTALLER_PATH="${OPTARG}";;
         m) PLUGIN_IMAGE="${OPTARG}";;
+	k) KUBECONFIG="${OPTARG}";;
```
Comment (danwinship) [inline hack/run-locally.sh]: change tabs to spaces everywhere

File: hack/run-locally.sh [hunk 4]
```diff
@@ -239,7 +254,13 @@ echo "rebuilding the CNO"
 hack/build-go.sh
 
 # Autodetect the state of the cluster to determine which mode to run in
-if [[ -z "$(ls -A ${CLUSTER_DIR} 2> /dev/null | grep -v install-config.yaml | grep -v .openshift_install | grep -v env.sh)" ]]; then
+if [[ ! -z "${KUBECONFIG}" ]]; then
```
Comment (danwinship) [inline hack/run-locally.sh]: I think if `CLUSTER_DIR` is set then we should use that by default. I often have a stray `KUBECONFIG` env var pointing to some no-longer-running cluster:

    danw@p50:~> echo $KUBECONFIG 
    /home/danw/Desktop/kubeconfig-1790652
    danw@p50:~> oc get nodes
    Unable to connect to the server: dial tcp: lookup api.ci-op-pdnwmn0m-endurance-a5b11.llc.devcluster.openshift.com on 10.11.5.19:53: no such host

File: hack/run-locally.sh [hunk 5]
```diff
@@ -239,7 +254,13 @@ echo "rebuilding the CNO"
...
+    export CLUSTER_DIR=$PWD
```
Comment (danwinship) [inline hack/run-locally.sh]: hm... what does this end up getting used for? Would it be better to use a temporary directory?

Comment (alexanderConstantinescu) [reply hack/run-locally.sh]: This is used for the `${CLUSTER_DIR}/env.sh` which contains the image overrides. I could use a temporary directory. Would `/tmp/env.sh` be fine?

Comment (danwinship) [reply hack/run-locally.sh]: Use `mktemp -d`.
Alternatively, write the environment to `${KUBECONFIG}.cno-env.sh` ?

Comment (alexanderConstantinescu) [reply hack/run-locally.sh]: Used `mktemp -d`

Comment (alexanderConstantinescu) [reply hack/run-locally.sh]: Acutally, I've been re-thinking this. Usually what we want is also to just create a `env.sh` and modify the image targeted, then re-deploy. 

I thus think it would be easier if we just use `CLUSTER_DIR=/tmp` if we only have a `$KUBECONFIG`. Then modify the `/tmp/env.sh` and re-deploy. I think it's easier as such.

**Comment by alexanderConstantinescu** [comment]
@danwinship Could you have a final look at this?Thanks in advance!
